ingestion/src/metadata/orm_profiler/profiler/core.py

```diff
@@ -383,14 +398,35 @@ def process(self, generate_sample_data: Optional[bool]) -> ProfilerResponse:
         logger.info(
             f"Computing profile metrics for {self.profiler_interface.table_entity.fullyQualifiedName.__root__}..."
         )
-        self.compute_metrics()
 
+        if (
+            not isinstance(self.table, DeclarativeMeta)
+            and self.table.serviceType == DatabaseServiceType.Datalake
+        ):
+            (
+                sample_data,
+                self.data_frame_list,
+            ) = self.profiler_interface.fetch_sample_data(self.table)
+            self.compute_metrics()
+            profile = self._check_profile_and_handle(self.get_profile())
+            table_profile = ProfilerResponse(
+                table=self.profiler_interface.table_entity,
+                profile=profile,
+                sample_data=sample_data,
+            )
+            return table_profile
```

Comment:
so the usual flow actually fetches the sample data, and then run compute_metrics while doing a query to the db and fetch the relevant metric calculation ( this is the sqa's compute metrics flow)

whereas the flow is a little different on computing metrics with sample data..
it fetches the sample data in chunks, return a list of chunks, run compute metrics on that same data.
using the same approach as sqa makes it time and memory consuming as it will have to fetch sample data twice.

Reply:
resolved